ledger/ledgercore/error.go

```diff
@@ -27,6 +27,17 @@ import (
 // ErrNoSpace indicates insufficient space for transaction in block
 var ErrNoSpace = errors.New("block does not have space for transaction")
 
+// Verify each custom error type implements the error interface, and declare which are pointer/value receivers.
+var (
+	_ error = (*TxnNotWellFormedError)(nil)
+	_ error = (*TransactionInLedgerError)(nil)
+	_ error = (*LeaseInLedgerError)(nil)
+	_ error = BlockInLedgerError{}
+	_ error = ErrNoEntry{}
```

Comment:
Does the linter use this to determine that you _shouldn't_ be using the pointer as an error?

Reply:
Yes, this is one of the heuristics it uses (and it also shows up in the Go codebase, I guess it is a convention) it also looks for value vs pointer receivers in the error interface implementation, but sometimes that doesn't match how it is returned, so this is prioritized above looking at method receivers